tests/component.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { afterEach, expect, test } from 'bun:test';
+import { cleanup, render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import { create } from 'zustand';
+import { createZustandContext } from '../src';
+import { allCats, newCat } from './mocks';
+
+type Cat = string;
+
+type CatStore = {
+	cats: Cat[];
+	addCat: (cat: Cat) => void;
+	removeCat: (cat: Cat) => void;
+};
+
+const [CatsProvider, useCatsStore] = createZustandContext((initialState: { cats: Cat[] }) =>
+	create<CatStore>((set) => ({
+		cats: initialState.cats,
+		addCat: (cat) => set((state) => ({ cats: [...state.cats, cat] })),
+		removeCat: (cat) => set((state) => ({ cats: state.cats.filter((c) => c !== cat) })),
+	})),
+);
+
+const CatsList = () => {
+	const cats = useCatsStore((state) => state.cats);
+	const addCat = useCatsStore((state) => state.addCat);
+	const removeCat = useCatsStore((state) => state.removeCat);
+
+	return (
+		<div>
+			<ul>
+				{cats.map((cat) => (
+					<li key={cat}>{cat}</li>
+				))}
+			</ul>
+			<button type="button" onClick={() => addCat(newCat)}>
+				Add new cat
+			</button>
+			<button type="button" onClick={() => removeCat(newCat)}>
+				Remove new cat
+			</button>
+		</div>
+	);
+};
+
+const renderApp = () =>
+	render(
+		<CatsProvider initialValue={{ cats: allCats }}>
+			<CatsList />
+		</CatsProvider>,
+	);
```

Comment:
would opt for repetition instead of this; it’s just indirection that saves us nothing and makes tests harder to grok